components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/util/ActionMgtAuditLoggerUtil.java

```diff
@@ -0,0 +1,324 @@
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.utils.AuditLog;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.jsonObjectToMap;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.triggerAuditLogEvent;
+
+/**
+ * Action management V2 audit logger util class.
+ */
+public class ActionMgtAuditLoggerUtil {
+
+    /**
+     * To get the current user, who is doing the current task.
+     *
+     * @return current logged-in user.
+     */
+    private static String getUser() {
+
+        String user = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isNotEmpty(user)) {
+            user = UserCoreUtil
+                    .addTenantDomainToEntry(user, CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+        } else {
+            user = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        return user;
+    }
+
+    /**
+     * Get the initiator for audit logs.
+     *
+     * @return Initiator id despite masking.
+     */
+    private static String getInitiatorId() {
+
+        String initiator = null;
+        String username = MultitenantUtils.getTenantAwareUsername(getUser());
+        String tenantDomain = MultitenantUtils.getTenantDomain(getUser());
+        if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(tenantDomain)) {
+            initiator = IdentityUtil.getInitiatorId(username, tenantDomain);
+        }
+        if (StringUtils.isBlank(initiator)) {
+            if (username.equals(CarbonConstants.REGISTRY_SYSTEM_USERNAME)) {
+                // If the initiator is wso2.system, the username need not be masked.
+                return LoggerUtils.Initiator.System.name();
+            }
+            initiator = LoggerUtils.getMaskedContent(getUser());
+        }
+        return initiator;
+    }
+
+    /**
+     * Print action audit log related to the operation.
+     *
+     * @param operation Operation associated with the state change.
+     * @param action    Action object to be logged.
+     */
+    public void printActionAuditLog(Operation operation, Action action) {
+
+        if (!LoggerUtils.isEnableV2AuditLogs()) {
+            return;
+        }
+        JSONObject data = createAuditLogEntry(action);
+        buildAuditLog(operation, data);
+    }
+
+    /**
+     * Print action audit log related to the operation by action ID.
+     *
+     * @param operation Operation associated with the state change.
+     * @param actionId  ID of the action to be logged.
+     * @param action    Action object to be logged.
+     */
+    public void printActionAuditLog(Operation operation, String actionId, Action action) {
+
+        if (!LoggerUtils.isEnableV2AuditLogs()) {
+            return;
+        }
+        JSONObject data = createAuditLogEntry(actionId, action);
+        buildAuditLog(operation, data);
+
```

Comment:
Remove this line

Reply:
Fixed with b5355995d46dbafa80bb825fa0da60e5bd5062ee